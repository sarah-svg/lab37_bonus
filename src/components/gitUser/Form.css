.Form {
    display: flex;
    width: 100px;
    flex-direction: column; 
}
.button {
    size: 2rem;
}