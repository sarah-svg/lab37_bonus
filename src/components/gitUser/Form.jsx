import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { findUser } from '../../actions/findUser';

import { searchUser } from '../../actions/searchUser';
import { setRepo } from '../../actions/setRepo';
import { gitHubUser } from '../../utilis/Fetch';
import styles from './Form.css';

const Form = () => {
  const dispatch = useDispatch();
  const search = useSelector(state => state.search);
  const handleChange = ({ target }) => {

    dispatch(searchUser(target.value));
  };
  const handleSubmit = async(e) => {
    e.preventDefault();
    console.log(search, 'form');
    const user = await gitHubUser(search);

    dispatch(findUser(user));
  };
  // eslint-disable-next-line keyword-spacing
  return(
    <form className={styles.Form} onSubmit={handleSubmit}>
      <label>
        Enter a user name on Github
        <textarea onChange={handleChange}
          placeholder="username">
        </textarea>
      </label>
      <button>Enter</button>
    </form>
  );

};
export default Form;


