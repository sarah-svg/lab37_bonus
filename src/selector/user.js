export const getUser = (state) => state.users.users;
export const getRepo = (state) => state.repos.repos;
