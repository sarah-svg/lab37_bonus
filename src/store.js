import { createStore } from 'redux';
import { composeWithDevTools } from 'rex-devtools-extension';
import reducer from './reducer/gitUserReducer';


const store = createStore(reducer, composeWithDevTools);

export default store;
