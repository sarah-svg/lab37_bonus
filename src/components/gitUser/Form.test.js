import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../store';
import Form from './Form';




describe('Form component', () => {
  afterEach(() => cleanup());
  it('renders UsernameForm', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Form />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
