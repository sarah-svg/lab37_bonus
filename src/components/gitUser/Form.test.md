/* eslint-disable max-len */
global.fetch = require('node-fetch');
import React from 'react';
import {  getByPlaceholderText,
  render, screen, waitFor } from '@testing-library/react';
import user from '@testing-library/user-event';
import Form from './Form';
import { MemoryRouter } from 'react-router-dom';

describe('form', () => {
  it('form', () => {
    render(<Form/>);
    const newUser = getByPlaceholderText('username');
    user.type(newUser);
    return waitFor(() => {
      expect(screen.getByText('Search for a Github user')).not.toBeEmptyDOMElement();
    });

  });
});
