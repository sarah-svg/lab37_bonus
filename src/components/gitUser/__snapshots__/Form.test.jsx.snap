// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders UsernameForm 1`] = `
<DocumentFragment>
  <form>
    <label>
      Enter a user name on Github
      <textarea
        placeholder="username"
      />
    </label>
    <button>
      Enter
    </button>
  </form>
</DocumentFragment>
`;
